Fix this test so that it succeeds when the library is not compiled with
SCOTCH_PTHREAD.

--- scotch_6.0.4/src/check/test_common_thread.c	2014-09-28 11:39:59.000000000 -0500
+++ scotch_6.0.4/src/check/test_common_thread.c	2015-01-10 00:52:00.076229542 -0600
@@ -175,14 +175,14 @@
 char *              argv[])
 {
   TestThreadGroup       groudat;
-#if ((defined COMMON_PTHREAD) || (defined SCOTCH_PTHREAD))
+#if ((defined COMMON_PTHREAD) && (defined SCOTCH_PTHREAD))
   TestThread * restrict thrdtab;
   int                   thrdnbr;
 #endif /* ((defined COMMON_PTHREAD) || (defined SCOTCH_PTHREAD)) */
 
   SCOTCH_errorProg (argv[0]);
 
-#if ((defined COMMON_PTHREAD) || (defined SCOTCH_PTHREAD))
+#if ((defined COMMON_PTHREAD) && (defined SCOTCH_PTHREAD))
   thrdnbr = SCOTCH_PTHREAD_NUMBER;
 
   groudat.redusum = COMPVAL (thrdnbr);
